runtime/oti/j9.h

```diff
@@ -309,6 +309,7 @@ static const struct { \
 	((NULL != (interfaceClass)) && ((J9_ITABLE_INDEX_UNRESOLVED != ((methodIndexAndArgCount) & ~255))))
 
 /* Macros for ValueTypes */
```

Comment:
I would add a new macro 

``` 
#define J9_CLASS_DISALLOWS_LOCKING (J9ClassIsValueType | J9ClassIsValueBased)
#define J9_DOES_CLASS_DISALLOW_LOCKING(clazz) J9_ARE_ALL_BITS_SET((clazz)->classFlags, J9_CLASS_DISALLOWS_LOCKING)
```

Reply:
The JIT will be able to use the `J9_CLASS_DISALLOWS_LOCKING` mask instead of the existing isValueType check

Reply:
Added new marcos, but naming them as `J9_CLASS_DISALLOWS_LOCKING_FLAGS`  and `J9_CLASS_ALLOW_LOCKING`

Reply:
Added new marcos, but naming them as J9_CLASS_DISALLOWS_LOCKING_FLAGS and J9_CLASS_ALLOW_LOCKING